### PR TITLE
Kicks Martial Arts out of the attack chain (yippee), makes it use signals, plus a large clean up of existing martial arts

### LINKED ISSUE #74829: Martial arts runtimes while on floor
## Round ID:

[#203986](https://scrubby.melonmesa.com/round/203986)

## Reproduction:

1. Give any character CQC
2. Lie on floor
3. Attempt to grab another character
4. Runtime appears

```[10:49:27] Runtime in atoms_movable.dm, line 1566: Cannot execute null.add traits().
proc name: setGrabState (/atom/movable/proc/setGrabState)
usr: Tilus/(Marlyn Compton)
usr.loc: (Primary Tool Storage (137,139,2))
src: Marlyn Compton (/mob/living/carbon/human)
src.loc: the floor (137,139,2) (/turf/open/floor/iron)
call stack:
Marlyn Compton (/mob/living/carbon/human): setGrabState(1)
Marlyn Compton (/mob/living/carbon/human): setGrabState(1)
CQC (/datum/martial_art/cqc): grab act(Marlyn Compton (/mob/living/carbon/human), Joe Fuchs (/mob/living/carbon/human))
Human (/datum/species/human): grab(Marlyn Compton (/mob/living/carbon/human), Joe Fuchs (/mob/living/carbon/human), CQC (/datum/martial_art/cqc))
Joe Fuchs (/mob/living/carbon/human): CtrlClick(Marlyn Compton (/mob/living/carbon/human))
Marlyn Compton (/mob/living/carbon/human): CtrlClickOn(Joe Fuchs (/mob/living/carbon/human))
Marlyn Compton (/mob/living/carbon/human): ClickOn(Joe Fuchs (/mob/living/carbon/human), "icon-x=15;icon-y=24;left=1;ctr...")
Joe Fuchs (/mob/living/carbon/human): Click(the floor (138,138,2) (/turf/open/floor/iron), "mapwindow.map", "icon-x=15;icon-y=24;left=1;ctr...")
Tilus (/client): Click(Joe Fuchs (/mob/living/carbon/human), the floor (138,138,2) (/turf/open/floor/iron), "mapwindow.map", "icon-x=15;icon-y=24;left=1;ctr...")```

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Kicks Martial Arts out of the attack chain. 
   - All Martial Arts attacks are now handled via unarmed attack or grab signals
   - This means all martial arts are now technically on the living level, allowing any mob that can unarmed attack to martial arts. Sort of. YMMV. 

- All martial arts block checking is now handled by the arts themselves, meaning you can selectively decide for a martial arts strike to not be blocked. Maybe good for the future. 

- A comprehensive cleanup of all existing martial arts. Improving var names, code, adding some missing animation calls, etc. 

Fixes #74829

## Why It's Good For The Game

Untangles the mess that is martial arts, making it a lot easier to work with the attack chain and making it overall a ton more consistent. 

## Changelog

:cl: Melbert
refactor: Big martial arts refactor, they should now overall act a ton more consistent. Also technically any mob can do martial arts. Let me know if something is funky. 
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
finally, some good PR content

File: code/datums/brain_damage/special.dm [hunk 1]
```diff
@@ -245,14 +245,15 @@
 
 /datum/brain_trauma/special/psychotic_brawling/on_gain()
 	..()
-	psychotic_brawling = new(null)
+	psychotic_brawling = new()
```
Comment (ZephyrTFA) [inline code/datums/brain_damage/special.dm:248]: new... what?
Please don't use implict new on var sets if the var is not a typed local var

**Comment by Jackal-boop** [comment]
I imagine we'll find something breaks from this. But this gives me a few ideas for future martial arts and maybe a martial arts based antag

File: code/datums/brain_damage/special.dm (hunk 1 above)
Comment (MrMelbert) [reply code/datums/brain_damage/special.dm:248]: wat? it's literally three lines above

**Review by ZephyrTFA — APPROVED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**